File: backend/src/main/java/com/smartcampus/model/Resource.java
package com.smartcampus.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "resources")
public class Resource {
    @Id
    private String id;
    private String name;
    private ResourceType type;
    private Integer capacity;
    private String location;
    private ResourceStatus status;
    private String description;

    public enum ResourceType {
        LECTURE_HALL, LAB, MEETING_ROOM, EQUIPMENT
    }

    public enum ResourceStatus {
        ACTIVE, OUT_OF_SERVICE
    }
}

File: implementation-mern/backend/models/Resource.js
const mongoose = require('mongoose');

const resourceSchema = new mongoose.Schema({
    name: {
        type: String,
        required: [true, 'Please add a resource name'],
        trim: true
    },
    type: {
        type: String,
        required: true,
        enum: ['Lecture Hall', 'Lab', 'Meeting Room', 'Equipment']
    },
    capacity: {
        type: Number,
        default: 0
    },
    location: {
        type: String,
        required: true
    },
    status: {
        type: String,
        required: true,
        enum: ['ACTIVE', 'OUT_OF_SERVICE'],
        default: 'ACTIVE'
    },
    availabilityWindows: [{
        day: String,
        startTime: String,
        endTime: String
    }],
    description: String
}, { timestamps: true });

module.exports = mongoose.model('Resource', resourceSchema);
